src/peergos/shared/user/CommittedWriterData.java

```diff
@@ -16,4 +18,18 @@ public CommittedWriterData(MaybeMultihash hash, WriterData props) {
     public String toString() {
         return hash.toString();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CommittedWriterData that = (CommittedWriterData) o;
+        return Objects.equals(hash, that.hash);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(hash);
```

Comment:
can just be hash.hashcode()

Reply:
I really miss non nullable types in Java

Reply:
fixed